ChemistryLib/PhreeqcKernel.cpp

```diff
@@ -166,6 +206,11 @@ void PhreeqcKernel::setAqueousSolutions(
                 // Set pH value by hydrogen concentration.
                 double const pH = -std::log10(concentration);
                 aqueous_solution.Set_ph(pH);
+
+                if (components.find("H(1)") != components.end())
```

Comment:
```suggestion
                if (auto it = components.find("H(1)"); it != components.end())
```
Then the second search is not needed.

Reply:
I don't understand where the second search takes place.

Reply:
in the if-condition, there is `compontes["H(1)"]` this invokes the second search.

Reply:
In fact, I do not understand why this is needed in the code. If I understood correctly this lines will only give the pH value according to the H(1) concentration, but the pH value is a mandatory input parameter. (The pH value in phreeqc is given in log (activity) instead of log (concentration).

Reply:
With this line, element `H` is considered as a component for the purpose of charge balance.

Reply:
Fixed.